### PR TITLE
CLN: display.notebook_repr_html -> display.html.notebook (GH11784)

### LINKED ISSUE #11784: display.notebook_repr_html -> display.html.notebook
reorg on the html options, which I think we are going to need some more e.g. `df.style`.

This will make it more consistent with the option config style

need to deprecate the original and add new.

### PR DESCRIPTION
**Pull request by GGordonGordon**
Deprecate display.notebook_repr_html option and move it to
display.html.notebook

- [X] closes #11784
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

File: pandas/core/config_init.py [hunk 1]
```diff
@@ -366,6 +371,8 @@ def table_schema_cb(key):
                        validator=is_int)
     cf.register_option('html.use_mathjax', True, pc_html_use_mathjax_doc,
                        validator=is_bool)
+    cf.register_option('html.notebook', True, pc_nb_repr_h_doc,
```
Comment (gfyoung) [inline pandas/core/config_init.py]: Let's rename the doc to `html_notebook_doc`.

File: pandas/tests/io/formats/test_format.py [hunk 2]
```diff
@@ -1651,6 +1651,12 @@ def test_period(self):
                "3 2013-04          2011-04  d")
         assert str(df) == exp
 
+    @tm.capture_stdout
+    def test_option_notebook_repr_html_deprecated(self):
+        with tm.assert_produces_warning(FutureWarning,
```
Comment (gfyoung) [inline pandas/tests/io/formats/test_format.py]: Reference issue number above.

**Comment by jorisvandenbossche** [comment]
I personally have some doubts about this deprecation. 1) a lot of people use this, just changing it for cosmetic reasons can be annoying, 2) I actually find the current name much more self-explanatory.

**Comment by jreback** [comment]
closing, @jorisvandenbossche logic is complling

**Pull request closed without merging**